File: src/SharedWorkspaceDetails/Compliance.js
import React from "react";
import "./Compliance.css";

const ComplianceItem = ({name, value, icon}) => (
    <div className={`Compliance-item ${value}`}>
        <i className={`fa fa-${icon}`} />
    </div>
);

const Compliance = ({workspace: {compliance: {pii_data, pci_data, phi_data}}}) => (
    <div className="Compliance">
        <ComplianceItem name="PHI data" value={phi_data} icon="stethoscope"/>
        <ComplianceItem name="PCI data" value={pci_data} icon="bank"/>
        <ComplianceItem name="PII data" value={pii_data} icon="info"/>
    </div>
);

export default Compliance;

File: common/src/main/scala/io/phdata/models/Compliance.scala
package io.phdata.models

import io.circe.{Decoder, Encoder}

case class Compliance(phiData: Boolean, pciData: Boolean, piiData: Boolean, id: Option[Long] = None)

object Compliance {

  val empty = Compliance(phiData = false, pciData = false, piiData = false)

  implicit final val encoder: Encoder[Compliance] =
    Encoder.forProduct3("phi_data", "pci_data", "pii_data")(c => (c.phiData, c.pciData, c.piiData))

  implicit final val decoder: Decoder[Compliance] =
    Decoder.forProduct3("phi_data", "pci_data", "pii_data")(
      (phi: Boolean, pci: Boolean, pii: Boolean) => Compliance(phi, pci, pii)
    )

}
